File: frontend/src/types/repair.ts
/** 维修系统前端类型定义 */

export type RepairIssueType =
  | 'plumbing'
  | 'appliance'
  | 'carpentry'
  | 'wall_floor'
  | 'plumbing_fixture'
  | 'other'

export type RepairStatus =
  | 'pending'
  | 'assigned'
  | 'in_progress'
  | 'completed'
  | 'confirmed'
  | 'rejected'
  | 'cancelled'

export type WorkerStatus = 'available' | 'working' | 'on_leave'

export interface RepairCreate {
  property_id: number
  issue_type: RepairIssueType
  description: string
  images?: string[]
  scheduled_time?: string
}

export interface RepairRead {
  id: number
  property_id: number
  tenant_id: number
  landlord_id: number
  assigned_worker_id: number | null
  issue_type: RepairIssueType
  description: string
  images: string[] | null
  status: RepairStatus
  scheduled_time: string | null
  completed_at: string | null
  work_record: string | null
  work_images: string[] | null
  created_at: string
  updated_at: string
  tenant_name: string | null
  landlord_name: string | null
  worker_name: string | null
  property_title: string | null
}

export type WorkerScope = 'platform' | 'apartment'

export const WORKER_SCOPE_LABELS: Record<WorkerScope, string> = {
  platform: '网站管理',
  apartment: '公寓管理',
}

export interface RepairWorker {
  id: number
  user_id: number
  manager_id: number
  status: WorkerStatus
  scope: WorkerScope
  skills: string[] | null
  phone: string
  total_jobs: number
  rating: number
  created_at: string
  username: string | null
}

export interface WorkerCreate {
  username: string
  password: string
  phone: string
  skills?: string[]
  scope?: WorkerScope
}

export interface WorkerStatusUpdate {
  status: WorkerStatus
}

export const ISSUE_TYPE_LABELS: Record<RepairIssueType, string> = {
  plumbing: '水电',
  appliance: '家电',
  carpentry: '门窗',
  wall_floor: '墙面地面',
  plumbing_fixture: '管道',
  other: '其他',
}

export const REPAIR_STATUS_LABELS: Record<RepairStatus, string> = {
  pending: '待处理',
  assigned: '已派单',
  in_progress: '维修中',
  completed: '待确认',
  confirmed: '已确认',
  rejected: '已拒绝',
  cancelled: '已取消',
}

export const REPAIR_STATUS_TAGS: Record<RepairStatus, string> = {
  pending: 'danger',
  assigned: 'warning',
  in_progress: '',
  completed: 'warning',
  confirmed: 'success',
  rejected: 'info',
  cancelled: 'info',
}

File: backend/app/schemas/repair.py
"""维修系统 Pydantic 验证模型"""
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.repair import RepairIssueType, RepairStatus, WorkerStatus


# ── 报修工单 ──────────────────────────────────────

class RepairCreate(BaseModel):
    """租客创建报修"""
    property_id: int
    issue_type: RepairIssueType
    description: str = Field(min_length=1, max_length=2000)
    images: list[str] | None = None
    scheduled_time: str | None = None  # 格式: "YYYY-MM-DD AM/PM"


class RepairUpdate(BaseModel):
    """房东/维修师傅更新工单"""
    status: RepairStatus | None = None
    assigned_worker_id: int | None = None
    work_record: str | None = None
    work_images: list[str] | None = None


class RepairRead(BaseModel):
    """工单详情"""
    id: int
    property_id: int
    tenant_id: int
    landlord_id: int
    assigned_worker_id: int | None = None
    issue_type: RepairIssueType
    description: str
    images: list[str] | None = None
    status: RepairStatus
    scheduled_time: str | None = None
    completed_at: str | None = None
    work_record: str | None = None
    work_images: list[str] | None = None
    created_at: str
    updated_at: str

    # 关联对象
    tenant_name: str | None = None
    landlord_name: str | None = None
    worker_name: str | None = None
    property_title: str | None = None

    model_config = {"from_attributes": True}


# ── 维修师傅 ──────────────────────────────────────

class WorkerCreate(BaseModel):
    """房东创建维修师傅"""
    username: str = Field(min_length=1, max_length=100)
    password: str = Field(min_length=6)
    phone: str = Field(min_length=1, max_length=32)
    skills: list[str] | None = None


class WorkerUpdate(BaseModel):
    """更新维修师傅信息"""
    phone: str | None = None
    skills: list[str] | None = None


class WorkerStatusUpdate(BaseModel):
    """调整维修师傅工作状态"""
    status: WorkerStatus


class WorkerRead(BaseModel):
    """维修师傅详情"""
    id: int
    user_id: int
    manager_id: int
    status: WorkerStatus
    skills: list[str] | None = None
    phone: str
    total_jobs: int
    rating: float
    created_at: str
    username: str | None = None  # 关联 User

    model_config = {"from_attributes": True}
